lib/vrl/compiler/src/target.rs

```diff
@@ -36,18 +37,18 @@ pub trait Target: std::fmt::Debug {
     /// When inserting into a coalesced path, the implementor is encouraged to
     /// insert into the right-most segment if none exists, but can return an
     /// error if needed.
-    fn insert(&mut self, path: &Path, value: Value) -> Result<(), String>;
+    fn insert(&mut self, path: &LookupBuf, value: Value) -> Result<(), String>;
 
     /// Get a value for a given path, or `None` if no value is found.
     ///
     /// See [`Object::insert`] for more details.
-    fn get(&self, path: &Path) -> Result<Option<Value>, String>;
+    fn get(&self, path: &LookupBuf) -> Result<Option<Value>, String>;
 
     /// Remove the given path from the object.
     ///
     /// Returns the removed object, if any.
     ///
     /// If `compact` is true, after deletion, if an empty object or array is
     /// left behind, it should be removed as well, cascading up to the root.
-    fn remove(&mut self, path: &Path, compact: bool) -> Result<Option<Value>, String>;
+    fn remove(&mut self, path: &LookupBuf, compact: bool) -> Result<Option<Value>, String>;
```

Comment:
Same here (and in other places), re: using `Lookup<'a>` instead of `&'a LookupBuf`.